subsys/caf/modules/ble_adv.c

```diff
@@ -498,23 +495,51 @@ static void ble_initialized(void)
 	started = true;
 }
 
+static inline void get_req_modules(struct module_flags *mf)
+{
+	BUILD_ASSERT(!IS_ENABLED(CONFIG_BT_SETTINGS) ||
+		     IS_ENABLED(CONFIG_CAF_SETTINGS_LOADER),
+		     "Bluetooth settings enabled, but settings loader module is disabled");
+
+	module_flags_set_bit(mf, MODULE_IDX(ble_state));
+#if CONFIG_CAF_SETTINGS_LOADER
+	module_flags_set_bit(mf, MODULE_IDX(settings_loader));
+#endif
+#if CONFIG_CAF_BLE_ADV_BLE_BOND_SUPPORTED
+	module_flags_set_bit(mf, MODULE_IDX(ble_bond));
+#endif
```

Comment:
Shouldn't we use `if IS_ENABLED(...)` instead of `#if ...`?

Reply:
Changed. I needed to double check to ensure that code is opt-out by compiler in that case to prevent linker errors.